src/test/java/org/gridsuite/shortcircuit/server/service/ShortCircuitServiceTest.java

```diff
@@ -1,179 +1,338 @@
+                objectMapper
+            );
+        } finally {
+            Mockito.reset(
+                notificationService,
+                uuidGeneratorService,
+                resultService,
+                parametersRepository,
+                objectMapper
+            );
+        }
+    }
 
-    @Mock
-    private VariantManager variantManager;
+    @Test
+    void testGetNonExistingParameters() {
+        final UUID pUuid = UUID.randomUUID();
+        when(parametersRepository.findById(any(UUID.class))).thenReturn(Optional.empty());
+        assertThat(shortCircuitService.getParameters(pUuid)).as("service call result").isEmpty();
+        verify(parametersRepository).findById(pUuid);
+    }
 
-    private ShortCircuitWorkerService workerService;
+    @Test
+    void testGetExistingParametersAndConversionToDto() {
+        final UUID pUuid = UUID.randomUUID();
+        final double minVoltDrop = new Random().nextDouble();
+        final AnalysisParametersEntity pEntity = spy(new AnalysisParametersEntity(pUuid, false, false, false, false, StudyType.STEADY_STATE, minVoltDrop,
+                ShortCircuitPredefinedConfiguration.ICC_MAX_WITH_CEI909, false, false, false, false, InitialVoltageProfileMode.NOMINAL));
+        when(parametersRepository.findById(any(UUID.class))).thenReturn(Optional.of(pEntity));
+        //can't spy call to fromEntity()
+        assertThat(shortCircuitService.getParameters(pUuid)).as("service call result")
+                .get().as("dto").usingRecursiveComparison().isEqualTo(new ShortCircuitParametersInfos(
+                        ShortCircuitPredefinedConfiguration.ICC_MAX_WITH_CEI909, new ShortCircuitParameters()
+                        .setWithLimitViolations(false)
+                        .setWithVoltageResult(false)
+                        .setWithFortescueResult(false)
+                        .setWithFeederResult(false)
+                        .setStudyType(StudyType.STEADY_STATE)
+                        .setMinVoltageDropProportionalThreshold(minVoltDrop)
+                        .setWithLoads(false)
+                        .setWithShuntCompensators(false)
+                        .setWithVSCConverterStations(false)
+                        .setWithNeutralPosition(false)
+                        .setInitialVoltageProfileMode(InitialVoltageProfileMode.NOMINAL)));
+        verify(pEntity).getPredefinedParameters();
+        verify(pEntity).isWithLimitViolations();
+        verify(pEntity).isWithVoltageResult();
+        verify(pEntity).isWithFortescueResult();
+        verify(pEntity).isWithFeederResult();
+        verify(pEntity).getStudyType();
+        verify(pEntity).getMinVoltageDropProportionalThreshold();
+        verify(pEntity).isWithLoads();
+        verify(pEntity).isWithShuntCompensators();
+        verify(pEntity).isWithVscConverterStations();
+        verify(pEntity).isWithNeutralPosition();
+        verify(pEntity, atLeastOnce()).getInitialVoltageProfileMode();
```

Comment:
Not sure this is really necessary

Reply:
It's not an important check, but necessary to pass `verifyNoMoreInteractions()`.